### ISSUE TITLE
No Performance Impact of Using `.init()` in Computed Properties

### ISSUE THREAD

**Issue by mineOz** [Trendyol/ios-guidelines#79]
### Description:
In the style guide, there are certain rules about the usage of `.init()`:

**Variable Declaration**: It is recommended to avoid using `.init()` when assigning values to variables.
**Nested Statements**: `.init()` usage is discouraged in nested statements.
**Convenience init**: It encourages using convenience initializers when a designated initializer is already present.
**Initializer Arguments**: Parameter names should not be omitted and should be consistent with the variable being initialized.
**Failable Initializers**: Failable initializers should be avoided whenever possible.

However, these rules do not seem to cover the case for computed properties. In computed properties, the return type is explicitly known by the compiler, so using `.init()` does not have a performance impact. The compiler doesn't need to infer the type, as it is already clearly defined.

**Suggestion**: I propose updating the style guide to allow the use of `.init()` in computed properties since there is no performance drawback in this context.

**Comment by aniltaskiran** [comment]
By the way, in my opinion, we don't need to care about build time that much. These type checks are not as effective as we thought.

**Comment by eyupgoymen** [comment]
Using .init doesn't contribute to performance in any way.

Regarding the import topic, we should import directly when necessary; using .init to avoid direct imports is incorrect and leads to hacky, improper workarounds.

Using .init in the code makes it difficult for Xcode to point us to the right places when we want to find the initializer of that struct. Additionally, adding new parameters to the initializer becomes more challenging.

In conclusion, to prevent unintended UIKit objects from ending up in the Presenter, .init should not be used. Avoiding .init helps make the initializer's parameters more visible and allows Xcode to locate them more easily within the application.
